File: TempCalc4/src/com/kve/tempcalc4/MainActivity.java
package com.kve.tempcalc4;

import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends ActionBarActivity  implements OnClickListener{

	private Button btnCalculate;
	private TextView tvAnswerValue;
	private EditText etValue;
	
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		if (savedInstanceState == null) 
		{
			tvAnswerValue = (TextView) 	findViewById(R.id.tvAnswerValue);
			etValue = 		(EditText)	findViewById(R.id.etValue);
			btnCalculate = 	(Button) 	findViewById(R.id.btnCalculate);

			btnCalculate.setOnClickListener(this);




		}
		//	else it should just figure itself out
	}


	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		// Handle action bar item clicks here. The action bar will
		// automatically handle clicks on the Home/Up button, so long
		// as you specify a parent activity in AndroidManifest.xml.
		int id = item.getItemId();
		if (id == R.id.action_settings) {
			return true;
		}
		return super.onOptionsItemSelected(item);
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.btnCalculate:
			Double fahr = Double.parseDouble(etValue.getText().toString());
			Double cent = ((fahr - 32.0)* (5.0/9.0));
			tvAnswerValue.setText("" + Math.round(cent));

			break;
		default:
			break;
		}		
	}

}

File: MyFavouriteRobot/app/src/main/java/com/sners/myfavouriterobot/MainActivity.kt
package com.sners.myfavouriterobot

import android.content.Context
import android.os.Bundle
import android.util.AttributeSet
import android.util.Log
import com.google.android.material.snackbar.Snackbar
import androidx.appcompat.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import com.sners.myfavouriterobot.data.MainViewModel
import com.sners.myfavouriterobot.data.Robot

import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        val viewModel = MainViewModel(application)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "You entered a name of ${editText.text}", Snackbar.LENGTH_LONG)
                .setAction("Action", null).show()
            val robot = Robot("Skania", "myfile",
                "The one and only Skania", "The world's greated house-bot",
                2000, 5)
            Log.i("RobotLogging", robot.toString())
        }
    }



    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
